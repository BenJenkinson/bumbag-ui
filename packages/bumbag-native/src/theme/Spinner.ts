export default {
  sizes: {
    small: '16px',
    default: '20px',
    medium: '28px',
    large: '36px',
  },
};
