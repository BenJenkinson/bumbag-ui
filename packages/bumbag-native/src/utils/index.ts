export * from './breakpoint';
export * from './getCSSFromStyleObject';
export * from './theme';
export * from './throttle';
export * from './useBreakpoint';
export * from './useStyle';
export * from './useTheme';
