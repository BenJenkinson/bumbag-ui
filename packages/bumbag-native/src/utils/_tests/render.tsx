import * as React from 'react';
import { ThemeConfig } from '../../types/theme';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (
  Component: any,
  { colorMode = 'default', theme = {} }: { colorMode?: string; theme?: ThemeConfig } = {}
) =>
  render(
    <Provider platform="native" theme={theme} colorMode={colorMode}>
      {Component}
    </Provider>
  );
