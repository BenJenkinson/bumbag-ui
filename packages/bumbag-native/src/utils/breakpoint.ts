import { Dimensions } from 'react-native';
import { theme } from './theme';

export function breakpoint(direction, values) {
  return (props) => {
    const breakpoints = theme('breakpoints')(props);
    const directionBreakpoints = breakpoints[direction];
    const size = Dimensions.get('window')[direction];

    const directionBreakpointsArray = Object.entries(directionBreakpoints).reverse();

    let hasMatched;
    let value = values.default;
    directionBreakpointsArray.forEach(([breakpoint, upperLimit], index) => {
      if (hasMatched) return;

      const lowerLimit = directionBreakpointsArray?.[index + 1]?.[1] || 0;

      // 'md', 'lg', etc takes first precedence
      if (size <= upperLimit && size > lowerLimit) {
        const matchedValue = values[breakpoint];
        value = matchedValue || value;
        hasMatched = Boolean(matchedValue);
      }

      // 'min-md', 'min-lg', etc takes second precedence
      if (size > lowerLimit) {
        const matcher = `min-${breakpoint}`;
        const matchedValue = values[matcher];
        value = matchedValue || value;
        hasMatched = Boolean(matchedValue);
      }

      // 'max-md', 'max-lg', etc takes third precedence
      if (size <= upperLimit) {
        const matcher = `max-${breakpoint}`;
        value = values[matcher] || value;
      }
    });

    return value;
  };
}
