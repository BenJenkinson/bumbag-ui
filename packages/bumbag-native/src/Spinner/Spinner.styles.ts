import { ActivityIndicator } from 'react-native';
import { styled } from '../styled';
import { theme } from '../utils/theme';

export const StyledSpinner = styled(ActivityIndicator)`
  display: inline-block;

  ${theme('native.Spinner', 'styles.base')};
`;
