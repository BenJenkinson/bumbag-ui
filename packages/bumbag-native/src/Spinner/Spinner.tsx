import { ActivityIndicatorProps as RNActivityIndicatorProps } from 'react-native';
import { createComponent, createElement, createHook, useTheme } from 'bumbag/utils';
import { Size } from 'bumbag/types';

import { palette, theme as getTheme } from '../utils/theme';
import { Box, BoxProps } from '../Box';
import * as styles from './Spinner.styles';

export type LocalSpinnerProps = {
  size?: Size;
};
export type SpinnerProps = BoxProps & Omit<RNActivityIndicatorProps, 'size'> & LocalSpinnerProps;

const useProps = createHook<SpinnerProps>(
  (props, { themeKey }) => {
    const { color, size, ...restProps } = props;
    const { theme } = useTheme();

    const boxProps = Box.useProps(restProps);

    return {
      ...boxProps,
      color: palette(color as string)({ theme }),
      size: getTheme(themeKey, `sizes.${size}`)({ theme })
        ? parseInt(getTheme(themeKey, `sizes.${size}`)({ theme }).replace('px', ''))
        : parseInt((size as string).replace('px', '')),
    };
  },
  { defaultProps: { color: 'primary', size: 'default' }, themeKey: 'native.Spinner' }
);

export const Spinner = createComponent<SpinnerProps>(
  (props) => {
    const htmlProps = useProps(props);
    return createElement({
      children: props.children,
      component: styles.StyledSpinner,
      htmlProps,
    });
  },
  {
    attach: {
      useProps,
      displayName: 'native.Spinner',
    },
    themeKey: 'native.Spinner',
  }
);
