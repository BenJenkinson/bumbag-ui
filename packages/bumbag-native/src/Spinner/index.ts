import * as spinnerStyles from './Spinner.styles';

export * from './Spinner';
export { spinnerStyles };
