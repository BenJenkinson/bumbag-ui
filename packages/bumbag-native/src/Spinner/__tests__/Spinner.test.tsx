import * as React from 'react';
import { applyTheme } from 'bumbag/utils/applyTheme';
import { Spinner } from '../Spinner';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Spinner />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with color', () => {
    const { container } = render(<Spinner color="primary" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  ['small', 'default', 'medium', 'large', '50px'].forEach((size) => {
    it(`should render correctly for size ${size}`, () => {
      const { container } = render(<Spinner size={size} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('Spinner.styles.base should render correctly', () => {
    const { container } = render(<Spinner />, {
      theme: { Spinner: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Spinner.styles.base should render correctly', () => {
    const { container } = render(<Spinner />, {
      theme: {
        Spinner: { styles: { base: (props) => ({ backgroundColor: props.color }) } },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('overrides', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Spinner
        overrides={{
          Spinner: {
            styles: { base: { backgroundColor: 'red' } },
            defaultProps: {
              altitude: '100',
              padding: 'major-2',
            },
          },
        }}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('applyTheme', () => {
  it('renders correctly', () => {
    const Card = applyTheme(Spinner, {
      styles: {
        base: {
          backgroundColor: 'red',
        },
      },
      defaultProps: {
        altitude: '100',
        padding: 'major-2',
      },
    });
    const { container } = render(<Card variant="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('variants', () => {
  it('styles.base should render correctly', () => {
    const { container } = render(<Spinner variant="test" />, {
      theme: {
        Spinner: {
          variants: { test: { styles: { base: { backgroundColor: 'red' } } } },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for overrides', () => {
    const { container } = render(
      <Spinner
        overrides={{
          Spinner: {
            variants: {
              test: {
                styles: { base: { backgroundColor: 'red' } },
                defaultProps: {
                  altitude: '100',
                  padding: 'major-2',
                },
              },
            },
          },
        }}
        variant="test"
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for applyTheme', () => {
    const Card = applyTheme(Spinner, {
      defaultProps: {
        altitude: '100',
        padding: 'major-2',
      },
      variants: {
        test: {
          defaultProps: {
            color: 'red',
          },
        },
      },
    });
    const { container } = render(<Card variant="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('modes', () => {
  it('should render correctly when colorMode is set globally', () => {
    const { container } = render(<Spinner />, {
      colorMode: 'test',
      theme: {
        Spinner: {
          modes: {
            test: {
              styles: { base: { backgroundColor: 'red' } },
              defaultProps: { color: 'primaryTint' },
            },
          },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly when colorMode set as prop', () => {
    const { container } = render(<Spinner colorMode="test" />, {
      theme: {
        Spinner: {
          modes: {
            test: {
              styles: { base: { backgroundColor: 'red' } },
              defaultProps: { color: 'primaryTint' },
            },
          },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for overrides', () => {
    const { container } = render(
      <Spinner
        overrides={{
          Spinner: {
            modes: {
              test: {
                styles: { base: { backgroundColor: 'red' } },
                defaultProps: {
                  altitude: '100',
                  padding: 'major-2',
                },
              },
            },
          },
        }}
        colorMode="test"
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly for applyTheme', () => {
    const Card = applyTheme(Spinner, {
      defaultProps: {
        altitude: '100',
        padding: 'major-2',
      },
      modes: {
        test: {
          styles: { base: { backgroundColor: 'red' } },
          defaultProps: { color: 'primaryTint' },
        },
      },
    });
    const { container } = render(<Card colorMode="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for color', () => {
    const { container } = render(<Spinner />, {
      theme: { Spinner: { defaultProps: { color: 'primary' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
