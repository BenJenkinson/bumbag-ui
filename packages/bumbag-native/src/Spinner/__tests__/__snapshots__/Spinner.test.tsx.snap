// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applyTheme renders correctly 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; background-color: rgb(239, 68, 68); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); justify-content: center; padding: 16px 16px 16px 16px;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 20px; width: 20px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`defaultProps should render correctly for color 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; justify-content: center;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 20px; width: 20px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`modes renders correctly for applyTheme 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; background-color: rgb(239, 68, 68); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); justify-content: center; padding: 16px 16px 16px 16px;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 20px; width: 20px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #eeedfd; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #eeedfd; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`modes renders correctly for overrides 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; background-color: rgb(239, 68, 68); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); justify-content: center; padding: 16px 16px 16px 16px;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 20px; width: 20px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`modes should render correctly when colorMode is set globally 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; background-color: rgb(239, 68, 68); justify-content: center;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 20px; width: 20px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #eeedfd; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #eeedfd; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`modes should render correctly when colorMode set as prop 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; background-color: rgb(239, 68, 68); justify-content: center;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 20px; width: 20px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #eeedfd; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #eeedfd; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`overrides should render correctly 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; background-color: rgb(239, 68, 68); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); justify-content: center; padding: 16px 16px 16px 16px;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 20px; width: 20px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`props should render correctly 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; justify-content: center;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 20px; width: 20px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`props should render correctly for size 50px 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; justify-content: center;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 50px; width: 50px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`props should render correctly for size default 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; justify-content: center;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 20px; width: 20px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`props should render correctly for size large 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; justify-content: center;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 36px; width: 36px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`props should render correctly for size medium 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; justify-content: center;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 28px; width: 28px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`props should render correctly for size small 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; justify-content: center;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 16px; width: 16px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`props should render correctly with color 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; justify-content: center;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 20px; width: 20px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`theming Spinner.styles.base should render correctly 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; background-color: rgb(239, 68, 68); justify-content: center;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 20px; width: 20px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`theming Spinner.styles.base should render correctly 2`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; background-color: rgb(87, 79, 235); justify-content: center;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 20px; width: 20px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`variants renders correctly for applyTheme 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; box-shadow: 0px 1px 4px rgba(0,0,0,0.15); justify-content: center; padding: 16px 16px 16px 16px;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 20px; width: 20px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #ef4444; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #ef4444; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`variants renders correctly for overrides 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; background-color: rgb(239, 68, 68); box-shadow: 0px 1px 4px rgba(0,0,0,0.15); justify-content: center; padding: 16px 16px 16px 16px;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 20px; width: 20px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;

exports[`variants styles.base should render correctly 1`] = `
<div
  aria-valuemax="1"
  aria-valuemin="0"
  class="css-view-1dbjc4n"
  role="progressbar"
  style="align-items: center; background-color: rgb(239, 68, 68); justify-content: center;"
>
  <div
    class="css-view-1dbjc4n r-animationKeyframes-127358a"
    style="animation-duration: 0.75s; animation-iteration-count: infinite; animation-timing-function: linear; height: 20px; width: 20px;"
  >
    <svg
      height="100%"
      viewBox="0 0 32 32"
      width="100%"
    >
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; opacity: 0.2;"
      />
      <circle
        cx="16"
        cy="16"
        fill="none"
        r="14"
        stroke-width="4"
        style="stroke: #574feb; stroke-dasharray: 80; stroke-dashoffset: 60;"
      />
    </svg>
  </div>
</div>
`;
